### ISSUE TITLE
First installation bootlooping "Starting web server on port: 80"

### ISSUE THREAD

**Issue by phillipse01** [s60sc/ESP32-Tuya_Device#1]
I followed the instructions and flashed my esp-c3f using Arduino IDE and it doesn't show up the WiFi. The serial output shows it crashing/looping as follows. (note, I added memory logging to see if it would help - it did not)

It crashes after **Starting web server on port: 80**

```
16:49:11.950 -> ESP-ROM:esp32c3-api1-20210207
16:49:11.950 -> Build:Feb  7 2021
16:49:11.996 -> rst:0x3 (RTC_SW_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
16:49:11.996 -> Saved PC:0x4038203e
16:49:11.996 -> SPIWP:0xee
16:49:11.996 -> mode:DIO, clock div:1
16:49:11.996 -> load:0x3fcd5810,len:0x438
16:49:11.996 -> load:0x403cc710,len:0x918
16:49:11.996 -> load:0x403ce710,len:0x25f4
16:49:11.996 -> entry 0x403cc710
16:49:12.228 -> 
16:49:12.228 -> [00:03:40.322 ramLogClear] Setup RAM based log
16:49:12.352 -> [00:03:40.324 listFolder] Sketch size 1320kB
16:49:12.352 -> [00:03:40.458 listFolder] LittleFS: Total bytes 131072, Used bytes 16384
16:49:12.352 -> startStorage > Free: heap 244164, block: 225268, pSRAM 0
16:49:12.570 -> [00:03:40.669 loadConfig] Load config
16:49:12.570 -> [00:03:40.670 ERROR @ prefs.cpp:133] Failed to load file /data/configs.txt 
16:49:12.570 -> [00:03:40.672 WARN retrieveConfigVal] Key hostName not set 
16:49:12.570 -> [00:03:40.682 WARN retrieveConfigVal] Key AP_SSID not set 
16:49:12.695 -> [00:03:40.832 setWifiSTA] Wifi Station IP from DHCP
16:49:12.741 -> setWifiSTA > Free: heap 203508, block: 180212, pSRAM 0
16:49:12.912 -> [00:03:41.039 onWiFiEvent] Wifi AP stopped: ESP-TuyaDevice_18AA0333E864
16:49:12.912 -> [00:03:41.040 onWiFiEvent] Wifi AP SSID: ESP-TuyaDevice_18AA0333E864 started, use 'http://192.168.4.1' to connect
16:49:12.959 -> setWifiAP > Free: heap 201396, block: 180212, pSRAM 0
16:49:13.143 -> [00:03:41.250 WARN startWifi] SSID  not available, use AP 
16:49:19.776 -> [00:03:47.884 startPing] Started ping monitoring
16:49:19.776 -> startPing > Free: heap 191564, block: 172020, pSRAM 0
16:49:19.950 -> [00:03:48.086 startWebServer] Starting web server on port: 80
16:49:19.996 -> startWebserver > Free: heap 182296, block: 163828, pSRAM 0
16:49:20.758 -> ESP-ROM:esp32c3-api1-20210207
16:49:20.758 -> Build:Feb  7 2021
16:49:20.802 -> rst:0x3 (RTC_SW_SYS_RST),boot:0xd (SPI_FAST_FLASH_BOOT)
16:49:20.802 -> Saved PC:0x4038203e
16:49:20.802 -> SPIWP:0xee
16:49:20.802 -> mode:DIO, clock div:1
16:49:20.802 -> load:0x3fcd5810,len:0x438
16:49:20.802 -> load:0x403cc710,len:0x918
16:49:20.802 -> load:0x403ce710,len:0x25f4
16:49:20.802 -> entry 0x403cc710
16:49:21.036 -> 
16:49:21.036 -> [00:03:49.134 ramLogClear] Setup RAM based log
16:49:21.161 -> [00:03:49.135 listFolder] Sketch size 1320kB
16:49:21.161 -> [00:03:49.269 listFolder] LittleFS: Total bytes 131072, Used bytes 16384
16:49:21.161 -> startStorage > Free: heap 244164, block: 225268, pSRAM 0
16:49:21.377 -> [00:03:49.481 loadConfig] Load config
16:49:21.377 -> [00:03:49.482 ERROR @ prefs.cpp:133] Failed to load file /data/configs.txt 
16:49:21.377 -> [00:03:49.483 WARN retrieveConfigVal] Key hostName not set 
16:49:21.377 -> [00:03:49.494 WARN retrieveConfigVal] Key AP_SSID not set 
16:49:21.502 -> [00:03:49.644 setWifiSTA] Wifi Station IP from DHCP
16:49:21.549 -> setWifiSTA > Free: heap 203508, block: 180212, pSRAM 0
16:49:21.753 -> [00:03:49.851 onWiFiEvent] Wifi AP stopped: ESP-TuyaDevice_18AA0333E864
16:49:21.753 -> [00:03:49.852 onWiFiEvent] Wifi AP SSID: ESP-TuyaDevice_18AA0333E864 started, use 'http://192.168.4.1' to connect
16:49:21.753 -> setWifiAP > Free: heap 201396, block: 180212, pSRAM 0
16:49:21.925 -> [00:03:50.063 WARN startWifi] SSID  not available, use AP 
16:49:28.401 -> [00:03:56.499 startPing] Started ping monitoring
16:49:28.401 -> startPing > Free: heap 190936, block: 172020, pSRAM 0
16:49:28.572 -> [00:03:56.702 startWebServer] Starting web server on port: 80
16:49:28.572 -> startWebserver > Free: heap 182296, block: 163828, pSRAM 0
16:49:29.382 -> ESP-ROM:esp32c3-api1-20210207
16:49:29.382 -> Build:Feb  7 2021

```
Any suggestions would help! Thanks!

**Comment by s60sc** [comment]
Could be [hardware issue](https://community.platformio.org/t/esp32-c3-continuously-reset/28568/12)

**Comment by phillipse01** [comment]
Hi @s60sc  - thanks for the link! I tried the simple loop (that prints got here) from that article to help see if it still crashed, however, I was unable to replicate the behavior - It looks like my hardware is working :) or maybe there is something not quite right with it? I am currently running the esp in the code burner before doing the swap

**Comment by phillipse01** [comment]
with trial and error I found the issue to only trigger when we get to the function in sniffer.cpp:
static void configureUart(uart_port_t uartNum) {

I was able to get the panic error dump, but unable to decode the back trace to anything useful. I did find however:

assert failed: xQueueReceive queue.c:1382 (( pxQueue ))

and based on https://esp32.com/viewtopic.php?f=13&t=34662 it looks like we are missing xQueueCreate() ?

**Comment by s60sc** [comment]
It will be something to do with your config. Could try adding LOG_ statements to pinpoint the problem.

**Comment by phillipse01** [comment]
Adding log lines, I can confirm the issue reliably and always occurs at:
https://github.com/s60sc/ESP32-Tuya_Device/blob/6a5ac18d0498efeb24a8f0702564acb6c7e3e585/sniffer.cpp#L173

with uartNum being 0

I tried adding xQueueCreate(..) to the function configureUart but it did not help :(

**Comment by s60sc** [comment]
uart 0 is in use if the c3 is in the burner, uart 0 can only be used when installed in a device

```
#define USE_SNIFFER true // false to be tuya device, true when sniffing and need to connect ESP to external wifi eg a removed WBR3 Tuya wifi module 
#define WIFI_TX_PIN 4 // only when USE_SNIFFER is true
#define WIFI_RX_PIN 3 // only when USE_SNIFFER is true
#define USE_UART0 false // true if using ESP as tuya device, else false if used as external wifi module
```

**Comment by phillipse01** [comment]
Hi s60sc

I was able to prevent the reboot/crash by adding:

```
uartQueue[0] = xQueueCreate (10, sizeof(unsigned int));
uartQueue[1] = xQueueCreate (10, sizeof(unsigned int));
```

to configureUart in sniffer.cpp - and now It is able to run!

Unfortunately I have a new error. When I run the command "M 8" I get a reply from the MCU but no response from Wifi (wBR3) chip:

```
[9:49:55 pm Web] Cmd: M8
[10:49:55.496 formatTuya] Wifi > [8]
[10:49:55.540 formatTuya] MCU > [7] DP 1: enum ( 1 )
[10:49:55.541 formatTuya] MCU > [7] DP 7: enum ( 0 )
```

And when I try W 3 command or any W command I get:
```
[9:50:29 pm Web] Cmd: W 3
[10:50:29.894 WARN processTuyaMsg] Uart 1 wrote -1, expected 7
```

Do you think there is any sensible defaults I should set for xQueueCreate that I am missing?

**Comment by s60sc** [comment]
You dont need to change the code, there is something wrong with your setup.

**Comment by phillipse01** [comment]
Thanks, however I have tried every config available in Arduino IDE - I also checked many articles and nothing stopped the xQueueReceive panic.

For reference on my setup, I did this:

-> Downloaded Arduino IDE fresh v2.2.1
-> Copied the zip of the repo to the sketch folder and unzipped - renamed (removed -main as per instructions)
-> installed esp32 boards by Espressif (v2.0.11)
-> ESP32C3 board in burner flashed with settings:
![image](https://github.com/s60sc/ESP32-Tuya_Device/assets/4907334/41da6ecd-1636-4d5e-a5c4-1d50b16aab57)

Without xQueueCreate it crashes immediately when calling xQueueReceive - nothing I do to any settings changes it :(

I just tried doing this same procedure fresh on my mac, same result with the crashes :( If you have any tips on where to look to make xQueueReceive work, I will be glad for the pointers! 

Thanks!
